src/bsoncxx/json.cpp

```diff
@@ -209,11 +209,11 @@ class json_visitor {
     std::vector<bool> stack;
     std::size_t padding;
 
-    void pad(std::size_t extra = 0) {
+    BSONCXX_PRIVATE void pad(std::size_t extra = 0) {
```

Comment:
More specifically: you should only need to add BSONCXX_PRIVATE or MONGOCXX_PRIVATE if the containing class or structure is marked MONGOCXX_API. Why?

We will build with -fvisibility=hidden, which makes *everything* hidden by default, unless marked with __attribute__((visibility("default"))), which is what MONGOCXX_API expands to.

The issue is that when you have a class that is tagged MONGOCXX_API, then all of its members become part of the ABI. By marking private members of *exported* classes as MONGOCXX_PRIVATE, we override the class wide export, and keep those functions from leaking into the ABI.

That way, if we need to change the signature, types, etc. in the private constructors or methods of these types, then those will not require an ABI bump.